__tests__/tests.js

```diff
@@ -527,4 +616,68 @@ describe('htmlbars-inline-precompile', function () {
       `);
     });
   });
+
+  describe('with transformScope: true', function () {
+    beforeEach(() => {
+      plugins = [
+        [
+          HTMLBarsInlinePrecompile,
+          {
+            precompile() {
+              return precompile.apply(this, arguments);
+            },
+
+            modules: {
+              '@ember/template-compilation': {
+                export: 'precompileTemplate',
+                shouldParseScope: true,
+              },
+            },
+          },
+        ],
+      ];
+    });
+
+    it('correctly handles scope', function () {
+      let source = 'hello';
+      transform(
+        `import { precompileTemplate } from '@ember/template-compilation';\nvar compiled = precompileTemplate('${source}', { scope: { foo, bar } });`
+      );
+
+      expect(optionsReceived).toEqual({
+        contents: source,
+        scope: ['foo', 'bar'],
+      });
+    });
+
+    it('errors if scope contains mismatched keys/values', function () {
+      expect(() => {
+        transform(
+          "import { precompileTemplate } from '@ember/template-compilation';\nvar compiled = precompileTemplate('hello', { scope: { foo: bar } });"
```

Comment:
FWIW, it doesn't seem obvious that we can't / shouldn't support this. We might circle back to it later and add support...

Reply:
Right, it's mainly just added complexity. I think we can consider supporting it in the future, but for now this makes things much simpler.